build.assets/version-check-prerelease/main.go

```diff
@@ -0,0 +1,45 @@
+/*
+We don't want to publish semver pre-release versions or versions with build
+metadata to package repositories, as these are typically internal builds for
+test purposes. -- 2021-12 Walt
+*/
+package main
+
+import (
+	"flag"
+	"log"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+func main() {
```

Comment:
`version-check-latest` and `version-check-prerelease` are almost identical. Would it be easier to just have one binary and a switch that swaps out your `check` func?

```
go run ./build.assets/version-check prerelease -tag ${DRONE_TAG}
```

And then in the code you'd just implement both `checkLatest(tag) error` and `checkPrerelease(tag) error` and select the appropriate check.

Reply:
:shrug: I considered a single binary approach and found a little duplication easier to manage.  My usual rule of thumb is to abstract the 3rd time I'm doing something and I only did this twice.

I don't feel strongly about this either way, and would be happy to change it if you do feel strongly.  Let me know.